app/src/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -623,8 +624,24 @@ protected void onActivityResult(int requestCode, int resultCode, Intent intent)
                 refreshCurrentView(false);
                 break;
         }
+        mFormController.setPendingCalloutFormIndex(null);
     }
-    
+
+    private IntentWidget tryToGetIntentWidget() {
+        IntentWidget bestMatch = null;
+
+        //Ugh, copied from the odkview mostly, that's stupid
+        for(QuestionWidget q : ((ODKView)mCurrentView).getWidgets()) {
+            //Figure out if we have a pending intent widget
+            if (q instanceof IntentWidget) {
+                if(((IntentWidget) q).isWaitingForBinaryData() || bestMatch == null) {
```

Comment:
roughly this should work as: q.getFormIndex.equals(getPendingFormIndex)

but yeah, specifically is concerning because the result of
tryToGetIntentWidget is used to control whether your new code is called at
all, which doesn't seem right

On Mon, Aug 10, 2015 at 10:55 AM, Daniel Roberts <notifications@github.com>
wrote:

> In app/src/org/odk/collect/android/activities/FormEntryActivity.java
> <https://github.com/dimagi/commcare-odk/pull/506#discussion_r36640191>:
>
> >      }
> > -
> > +
> > +    private IntentWidget tryToGetIntentWidget() {
> > +        IntentWidget bestMatch = null;
> > +
> > +        //Ugh, copied from the odkview mostly, that's stupid
> > +        for(QuestionWidget q : ((ODKView)mCurrentView).getWidgets()) {
> > +            //Figure out if we have a pending intent widget
> > +            if (q instanceof IntentWidget) {
> > +                if(((IntentWidget) q).isWaitingForBinaryData() || bestMatch == null) {
>
> I see, you mean reimplement this function as
>
> return
> mFormController.getPendingCalloutFormIndex().whateverTheGetCurrentWidgetFunctionIsCalled()
>
> right? I wasn't certain that was possible, but your hinting that it is
> gives me courage so I'll give it a shot.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dimagi/commcare-odk/pull/506/files#r36640191>.
>